protocol/contracts/nexus/TempleSacrifice.sol

```diff
@@ -0,0 +1,71 @@
+//                                       @@@
+//                                       @@@
+//                                       @@@
+//                                       @@@
+//                                       @@@
+//                                       @@@
+//                                       @@@
+//               @@@                    @@@@@                    @@@
+//               @@@                  @@@@@@@@@                  @@@
+//               @@@               .@@@@@@@@@@@@@.               @@@
+//               @@@             (@@@@@@@@@@@@@@@@@)             @@@
+//               @@@           @@@@@@@@@@@@@@@@@@@@@@@           @@@
+//               @@@         @@@@@@@@@@@@@@@@@@@@@@@@@@@         @@@
+//               @@@       @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@       @@@
+//               @@@    .@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@.    @@@
+//               @@@  /@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@\  @@@
+//               @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+//               @@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IRelic {
+    function whitelistTemplar(address _toWhitelist) external;
+}
+
+contract TempleSacrifice is Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    IRelic public RELIC;
+    IERC20 public TEMPLE;
+    uint256 private originTime;
+    uint256 public customPrice;
+
+    function sacrifice() external nonReentrant {
```

Comment:
What is reentrant about this function? I don't think `nonReentrant` is required?